editor/debugger/debug_adapter/debug_adapter_protocol.cpp

```diff
@@ -116,12 +116,12 @@ Error DAPeer::send_data() {
 		if (!data.has("seq")) {
 			data["seq"] = ++seq;
 		}
-		String formatted_data = format_output(data);
+		const Vector<uint8_t> &formatted_data = format_output(data);
 
 		int data_sent = 0;
-		while (data_sent < formatted_data.length()) {
+		while (data_sent < formatted_data.size()) {
 			int curr_sent = 0;
-			Error err = connection->put_partial_data((const uint8_t *)formatted_data.utf8().get_data(), formatted_data.size() - data_sent - 1, curr_sent);
+			Error err = connection->put_partial_data(formatted_data.ptr(), formatted_data.size() - data_sent, curr_sent);
```

Comment:
I assume null termination is not necessary, since it had `-1` before.

Reply:
For the reference, the same code for GDScript do add it:

https://github.com/godotengine/godot/blob/1f56d96cf2c768c3844c68ccb504dfeee841ae15/modules/gdscript/language_server/gdscript_language_protocol.cpp#L107-L114

Reply:
Fixed; I did a "hack" to test this scenario, and it behaves as expected.
![image](https://github.com/user-attachments/assets/b8d672d3-8ffd-47b4-b886-c1f7a96f0c04)
![image](https://github.com/user-attachments/assets/fc0e5ecf-d47b-4add-98e1-99a0771a9893)
![image](https://github.com/user-attachments/assets/0e460412-f679-4a4f-a041-3e2ddf8a5862)